relay-event-normalization/src/normalize/span/ai.rs

```diff
@@ -0,0 +1,97 @@
+//! AI cost calculation.
+
+use crate::ModelCosts;
+use relay_base_schema::metrics::MetricUnit;
+use relay_event_schema::protocol::{Event, Measurement, Span};
+
+/// Calculated cost is in US dollars.
+fn calculate_ai_model_cost(
+    model_id: &str,
+    prompt_tokens_used: Option<f64>,
+    completion_tokens_used: Option<f64>,
+    total_tokens_used: Option<f64>,
+    ai_model_costs: &ModelCosts,
+) -> Option<f64> {
+    if let Some(prompt_tokens) = prompt_tokens_used {
+        if let Some(completion_tokens) = completion_tokens_used {
+            let mut result = 0.0;
+            if let Some(cost_per_1k) = ai_model_costs.cost_per_1k_tokens(model_id, false) {
+                result += cost_per_1k * (prompt_tokens / 1000.0)
+            }
+            if let Some(cost_per_1k) = ai_model_costs.cost_per_1k_tokens(model_id, true) {
+                result += cost_per_1k * (completion_tokens / 1000.0)
+            }
```

Comment:
nit: There's an opportunity to call `cost_per_1k_tokens` only once (and iterate the list only once) if we make it return a tuple `(/*prompt tokens*/, Option<f32>, Option<f32> /*completion tokens*/)`. But I'm not sure if it increases readability.

Reply:
I don't think that's really better, will leave it as-is